### ISSUE TITLE
Pagerduty 'details' under template lose their order

### ISSUE THREAD

**Issue by efekete** [prometheus-operator/prometheus-operator#4668]
**What happened?**

Prometheus Pagerduty template details fields loses order

**Did you expect to see something different?**

Prometheus Pagerduty template fields  fields to keep the same order

**How to reproduce it (as minimally and precisely as possible)**:

Create pagerduty template with for example five fields. 
Apply it to Kubernetes. 
After operator generates configuration that will be used in Alertmanager, fields lose their order
Pagerduty alert table has wrong order of fields

* Prometheus Operator version:
 quay.io/prometheus-operator/prometheus-operator:v0.54.1

I configured alertmanger and pagerduty to work together. I would like to order details in the template so when alert comes to pagerduty, I can have namespace first, description second, etc.
Problem is, when operator generates configuration, it reorders the values. My guess is it happens cause of Go “built in feature” when looping through map it always goes random.
Here is the code where it happens:
https://github.com/prometheus-operator/prometheus-operator/blob/f49dc4bb80e33f0044df4915ab4e844feb30c02e/pkg/alertmanager/amcfg.go#L668

**Comment by simonpasquier** [comment]
> Problem is, when operator generates configuration, it reorders the values. My guess is it happens cause of Go “built in feature” when looping through map it always goes random.

This is correct. Even if the operator was able to serialize the map with a stable order for keys, it would only be reflected in the generated Alertmanager configuration. I assume that it's more a question for PagerDuty/Alertmanager than the operator itself.

**Comment by efekete** [comment]
Hello @simonpasquier,

Thank you for the fast reply.

It seems to me Alertmanager keeps the configuration same as it is in `/etc/alertmanager/config/alertmanager.yaml`, it only adds some default values as "firing". Also, pagerduty table order are the same as the one in configuration. Didn't check the code, but my guess is they are only using structs for marshal/unmarshal and it keeps the order. 

Anyway, if we manage to keep the order on alertmanager, can we do the same on operator side?

**Comment by efekete** [comment]
To get relevant information first when reading it on the phone

**Comment by simonpasquier** [comment]
This would be more a question for Alertmanager. But checking on the PagerDuty community forum, it seems that depending on the number of items, the`custom_details` keys are ordered either alphabetically or randomly hence it shouldn't matter if in which order they appear in the Alertmanager configuration?
https://community.pagerduty.com/forum/t/custom-details-field-ordering-and-whitespace/3671/4

**Comment by efekete** [comment]
Thx for the help! I'll close this issue then.
